### PR TITLE
Remove unnecessary calls to ASR utils

### PR DESCRIPTION
**Pull request by namannimmo10**

### CODE REVIEW

File: tests/constants1.py [hunk 1]
```diff
@@ -35,12 +35,13 @@ def test_len():
...
-    a = len([[-4, -5, -6], [-1, -2, -3]])
-    a = len({1, 2, 3})
-    a = len({1: "a", 2: "b", 3: "c"})
+    # TODO(namannimmo10): These commented out tests should work
+    # a = len((1, 2, 3))
+    # a = len((("c", "b", 3.4), ("c", 3, 5.6)))
+    # a = len([1, 2, 3])
+    # a = len([[-4, -5, -6], [-1, -2, -3]])
+    # a = len({1, 2, 3})
+    # a = len({1: "c", 2: "b", 3: "c"})
```
Comment (certik) [inline tests/constants1.py:44]: Why don't they work anymore?

File: src/lpython/semantics/python_comptime_eval.h [hunk 2]
```diff
@@ -131,12 +131,13 @@ struct PythonIntrinsicProcedures {
     }
 
     static ASR::expr_t *eval_str(Allocator &al, const Location &loc, Vec<ASR::expr_t*> &args) {
+        LFORTRAN_ASSERT(ASRUtils::all_args_evaluated(args));
```
Comment (certik) [inline src/lpython/semantics/python_comptime_eval.h:134]: This assert might stay --- once we allow unevaluated arguments that we seem to need for some intrinsic functions, then the callback can sometimes have arguments which are not evaluated. So having this asssert statement there is a good idea, to catch bugs early.

File: tests/constants1.py (hunk 1 above)
Comment (namannimmo10) [reply tests/constants1.py:44]: Looks like the argument is not a constant, and returns false when it runs across...

https://github.com/lcompilers/lpython/blob/2dd1e4ddd3e40ed65f2db9db3fe30c1247ef41dc/src/libasr/asr_utils.h#L417-L466

File: src/lpython/semantics/python_comptime_eval.h (hunk 2 above)
Comment (certik) [reply src/lpython/semantics/python_comptime_eval.h:134]: I made a mistake, I thought this was removing the assert. Yes, adding the asserts is a good idea.

**Review by certik — APPROVED** [review]
This looks good to merge.

**Pull request merged**